MekHQ/src/mekhq/gui/adapter/TOEMouseAdapter.java

```diff
@@ -810,13 +806,25 @@ private void maybeShowPopup(MouseEvent e) {
                             weightClassForUnitType.get(weightClassMenuName).setEnabled(false);
                         }
                     }
-                    
+
+                    for (int wc = EntityWeightClass.WEIGHT_SMALL_SUPPORT; wc <= EntityWeightClass.WEIGHT_LARGE_SUPPORT; wc++) {
+                        for (int ut : svTypes) {
+                            String typeName = UnitType.getTypeName(ut);
+                            String wcName = EntityWeightClass.getClassName(wc, typeName, true);
+                            String menuName = typeName + "_" + wcName;
```

Comment:
Does the compiler get it, or is it more efficient to use string.format for string concatenation?

Reply:
I used the same approach that's used in the surrounding code. Internally concatenation uses StringBuilder, which is significantly more efficient than String::format. String::format also introduces the possibility of an IllegalArgumentException at runtime, though a single test run will show it to work correctly in this case. I tend to use String::format when the format string itself is pulled from a resource file or in cases when it makes the code more readable and it's worth losing a nanosecond or so.

Reply:
Gotcha, guess I have C# on my brain.